site/src/pages/ManagementSettingsPage/GroupsPage/GroupPage.tsx

```diff
@@ -0,0 +1,361 @@
+  MoreMenuContent,
+  MoreMenuItem,
+  MoreMenuTrigger,
+  ThreeDotsButton,
+} from "components/MoreMenu/MoreMenu";
+import {
+  PageHeader,
+  PageHeaderSubtitle,
+  PageHeaderTitle,
+} from "components/PageHeader/PageHeader";
+import { Stack } from "components/Stack/Stack";
+import {
+  PaginationStatus,
+  TableToolbar,
+} from "components/TableToolbar/TableToolbar";
+import { UserAutocomplete } from "components/UserAutocomplete/UserAutocomplete";
+import { UserAvatar } from "components/UserAvatar/UserAvatar";
+import { isEveryoneGroup } from "utils/groups";
+import { pageTitle } from "utils/page";
+
+export const GroupPage: FC = () => {
+  const { organization, groupName } = useParams() as {
+    organization: string;
+    groupName: string;
+  };
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const groupQuery = useQuery(group(organization, groupName));
+  const groupData = groupQuery.data;
+  const { data: permissions } = useQuery(
+    groupData !== undefined
+      ? groupPermissions(groupData.id)
+      : { enabled: false },
+  );
+  const addMemberMutation = useMutation(addMember(queryClient));
+  const deleteGroupMutation = useMutation(deleteGroup(queryClient));
+  const [isDeletingGroup, setIsDeletingGroup] = useState(false);
+  const isLoading = groupQuery.isLoading || !groupData || !permissions;
+  const canUpdateGroup = permissions ? permissions.canUpdateGroup : false;
+
+  const helmet = (
+    <Helmet>
+      <title>
+        {pageTitle(
+          (groupData?.display_name || groupData?.name) ?? "Loading...",
+        )}
+      </title>
+    </Helmet>
+  );
+
+  if (groupQuery.error) {
+    return <ErrorAlert error={groupQuery.error} />;
+  }
+
+  if (isLoading) {
+    return (
+      <>
+        {helmet}
+        <Loader />
+      </>
+    );
+  }
+  const groupId = groupData.id;
+
+  return (
+    <>
+      {helmet}
+
+      <Margins>
+        <PageHeader
+          actions={
+            canUpdateGroup && (
+              <>
+                <Button
+                  startIcon={<SettingsOutlined />}
+                  to="settings"
+                  component={RouterLink}
+                >
+                  Settings
+                </Button>
+                <Button
+                  disabled={groupData?.id === groupData?.organization_id}
+                  onClick={() => {
+                    setIsDeletingGroup(true);
+                  }}
+                  startIcon={<DeleteOutline />}
+                  css={styles.removeButton}
+                >
+                  Delete&hellip;
+                </Button>
+              </>
+            )
+          }
+        >
+          <PageHeaderTitle>
+            {groupData?.display_name || groupData?.name}
+          </PageHeaderTitle>
+          <PageHeaderSubtitle>
+            {/* Show the name if it differs from the display name. */}
+            {groupData?.display_name &&
+            groupData?.display_name !== groupData?.name
```

Comment:
nit: I noticed that this logic is used in multiple places. Would it make sense to extract it into a utility function to ensure consistency?

Reply:
I think it'd probably make more sense as a component, but we can try that